src/ensembl/src/content/app/species/state/general/speciesGeneralHelper.ts

```diff
@@ -0,0 +1,462 @@
+    title: 'Assembly',
+    exampleLinkText: 'Example region',
+    groups: [Groups.ASSEMBLY_STATS],
+    primaryStatsKey: Stats.CHROMOSOMES
+  },
+  [SpeciesStatsSection.PSEUDOGENES]: {
+    title: 'Pseudogenes',
+    groups: [Groups.PSEUDOGENES],
+    primaryStatsKey: Stats.PSEUDOGENES
+  },
+  [SpeciesStatsSection.NON_CODING_STATS]: {
+    title: 'Non-coding genes',
+    groups: [Groups.NON_CODING_GENES],
+    primaryStatsKey: Stats.NON_CODING_GENES
+  }
+};
+
+// Maps the individual stats to it's respective groups
+const groupsStatsMap = {
+  [Groups.CODING_GENES]: [
+    [Stats.CODING_GENES, Stats.SHORTEST_GENE_LENGTH, Stats.LONGEST_GENE_LENGTH]
+  ],
+  [Groups.ANALYSIS]: [
+    [Stats.AVERAGE_GENOMIC_SPAN, Stats.AVERAGE_SEQUENCE_LENGTH],
+    [
+      Stats.AVERAGE_CDS_LENGTH,
+      Stats.TOTAL_TRANSCRIPTS,
+      Stats.CODING_TRANSCRIPTS,
+      Stats.TRANSCRIPTS_PER_GENE,
+      Stats.CODING_TRANSCRIPTS_PER_GENE,
+      Stats.TOTAL_EXONS,
+      Stats.TOTAL_CODING_EXONS,
+      Stats.AVERAGE_EXON_LENGTH,
+      Stats.AVERAGE_CODING_EXON_LENGTH,
+      Stats.AVERAGE_EXONS_PER_TRANSCRIPT,
+      Stats.AVERAGE_CODING_EXONS_PER_CODING_TRANSCRIPT,
+      Stats.TOTAL_INTRONS,
+      Stats.AVERAGE_INTRON_LENGTH
+    ]
+  ],
+  [Groups.ASSEMBLY_STATS]: [
+    [
+      Stats.CHROMOSOMES,
+      Stats.TOTAL_GENOME_LENGTH,
+      Stats.TOTAL_CODING_SEQUENCE_LENGTH,
+      Stats.AVERAGE_GC_CONTENT,
+      Stats.TOTAL_GAP_LENGTH,
+      Stats.SPANNED_GAP,
+      Stats.TOPLEVEL_SEQUENCES,
+      Stats.COMPONENT_SEQUENCES,
+      Stats.ASSEMBLY_NAME,
+      Stats.ASSEMBLY_DATE,
+      Stats.ASSEMBLY_ACCESSION,
+      Stats.CONTIG_N50,
+      Stats.TAXONOMY_ID,
+      Stats.STRAIN,
+      Stats.SEX,
+      Stats.SCIENTIFIC_NAME
+    ]
+  ],
+  [Groups.PSEUDOGENES]: [[Stats.PSEUDOGENES]],
+  [Groups.NON_CODING_GENES]: [
+    [
+      Stats.NON_CODING_GENES,
+      Stats.SMALL_NON_CODING_GENES,
+      Stats.LONG_NON_CODING_GENES,
+      Stats.MISC_NON_CODING_GENES
+    ]
+  ]
+};
+
+// Individual stat formatting options.
+type StatsFormattingOptions = {
+  [key in Stats]: {
+    label: string;
+    headerUnit?: string;
+    primaryUnit?: string;
+    secondaryUnit?: string;
+    primaryValuePostfix?: string;
+  };
+};
+
+const statsFormattingOptions: StatsFormattingOptions = {
+  [Stats.CODING_GENES]: {
+    label: 'Coding genes'
+  },
+  [Stats.SHORTEST_GENE_LENGTH]: {
+    label: 'Shortest coding gene',
+    primaryUnit: 'bp'
+  },
+  [Stats.LONGEST_GENE_LENGTH]: {
+    label: 'Longest coding gene',
+    primaryUnit: 'bp'
+  },
+  [Stats.CHROMOSOMES]: {
+    label: 'Chromosomes or plasmids',
+    headerUnit: 'chromosomes or plasmids'
+  },
+  [Stats.TOTAL_GENOME_LENGTH]: {
+    primaryUnit: 'bp',
+    label: 'Total genome length'
+  },
+  [Stats.TOTAL_CODING_SEQUENCE_LENGTH]: {
+    primaryUnit: 'bp',
+    label: 'Total coding sequence length'
+  },
+  [Stats.AVERAGE_GC_CONTENT]: {
+    primaryValuePostfix: '%',
```

Comment:
what's the difference between `primaryUnit` and `primaryValuePostfix`?

Reply:
I've removed the need to `primaryValuePostfix` as it's simpler to append the postfix value in the helper function itself.